src/compiler/crystal/codegen/debug.cr

```diff
@@ -186,7 +186,7 @@ module Crystal
       # any case, macro code *should* be simple so that it doesn't
       # need to be debugged at runtime (because macros work at compile-time.)
       unless filename.is_a?(String)
-        raise "BUG: expected debug filename to be a String, not #{filename.class}"
+        unreachable! "expected debug filename to be a String, not #{filename.class}"
```

Comment:
I checked this code and all points calling `file_and_dir`, then I think here is unreachable surely. `file_and_dir` is called by 5 points: L12, L157, L226, L259, L270. L12 passes `String` to `file_and_dir` directly. L259 passes `filename` variable, which is typed `String` for now. L157, L226 and L270 pass `location.filename`, however this `location` comes from `location.original_location`. `original_location.filename` is always `String`.